azure/scope/managedcontrolplane.go

```diff
@@ -270,14 +270,23 @@ func (s *ManagedControlPlaneScope) Vnet() *infrav1.VnetSpec {
 
 // GroupSpecs returns the resource group spec.
 func (s *ManagedControlPlaneScope) GroupSpecs() []azure.ASOResourceSpecGetter[*asoresourcesv1.ResourceGroup] {
-	return []azure.ASOResourceSpecGetter[*asoresourcesv1.ResourceGroup]{
+	specs := []azure.ASOResourceSpecGetter[*asoresourcesv1.ResourceGroup]{
 		&groups.GroupSpec{
 			Name:           s.ResourceGroup(),
 			Location:       s.Location(),
 			ClusterName:    s.ClusterName(),
 			AdditionalTags: s.AdditionalTags(),
 		},
 	}
+	if s.Vnet().ResourceGroup != "" && s.Vnet().ResourceGroup != s.ResourceGroup() {
+		specs = append(specs, &groups.GroupSpec{
+			Name:           s.Vnet().ResourceGroup,
```

Comment:
One thing I ran into trying to adopt an existing AKS cluster is that by default, AKS creates and manages a vnet in the node (`MC_*`) resource group, and Kubernetes doesn't allow names to contain underscores, so we would need separate names for the ASO resource and the Azure resource in that case.

I'll check to see if this is in scope for the adoption issue (#1173) but even if it's not, I think it's worth specifying separate names here. Then following through to see where we might be assuming those names are the same and changing those places accordingly.

Reply:
>  I think it's worth specifying separate names here. Then following through to see where we might be assuming those names are the same and changing those places accordingly.

Can please you elaborate more on this? 
`s.Vnet().ResourceGroup` is the name of the resource group created by the user in pre-existing VNet scenario.

Reply:
I mean the difference between `metadata.name` and `spec.azureName` in the ASO resource group for the vnet.

Reply:
So, are you saying that since existing _AzureName_ can have `_` and since Kubernetes won't tolerate it, we should be setting `s.Vnet().ResourceGroup` to the `AzureName` in ASO resources and have a normalized(without `_`) names set to the metadata.name for the ASO resource?

Reply:
I will add a soft conversion and push the changes.

Reply:
If we are looking to achieve something similar to this 20c05ee8242d298999ac71d828f82e3f1bdcdab6, I'd vote to address it via a different PR since metadata.name and AzureName changes will affect most of the resources.

Reply:
> So, are you saying that since existing _AzureName_ can have `_` and since Kubernetes won't tolerate it, we should be setting `s.Vnet().ResourceGroup` to the `AzureName` in ASO resources and have a normalized(without `_`) names set to the metadata.name for the ASO resource?

Yes. I'm not suggesting we make this change everywhere though, since there's a risk in changing a name during an upgrade which would recycle those ASO resources disruptively. Since we're close to a release, I'd strongly prefer to make that change here in this PR and not a follow-up to reduce the risk that we have to make this change after users might already be using it.

Reply:
> I'm not suggesting we make this change everywhere though

Ideally, let's try to scope this change to only this resource group for the vnet and no other resources.

Reply:
> Since we're close to a release, I'd strongly prefer to make that change here in this PR and not a follow-up to reduce the risk that we have to make this change after users might already be using it.

I am still hesitant on making the naming change in this PR. I agree that we should be normalizing the name, but better if we address it via different PR since it will be easier to revert if it creates side effects.

Reply:
There won't be any side effects if we limit the change to this one resource which wasn't being created before anyway.

Reply:
You could also still keep the name change in a separate commit in this same PR which would make reverting just as easy if we need to.

Reply:
Different commit achieves the same, will keep a different commit for it. I am not sure if k8s-ci-bot squash merges the commits. Do you know ?

Reply:
It won't squash anything automatically unless we tell it to with a label.